Source/UI/CommonUI/View/Component/InputView.swift

```diff
@@ -216,9 +222,11 @@ extension InputView: TextFieldViewDelegate {
     func textFieldShouldEndEditing(textField: UITextField, replacementString: String) -> Bool {
         let shouldEndEditing = delegate?.textFieldShouldEndEditing(textField: textField, replacementString: replacementString) ?? true
         if shouldEndEditing {
-            textFieldContainerBorder.layer.borderColor = (style?.error?.isHidden ?? true) ?
-            style?.textfield.borderStyle.normalColor.cgColor :
-                style?.textfield.borderStyle.errorColor.cgColor
+            let normalColor = style?.textfield.borderStyle.normalColor
+            let errorColor = style?.textfield.borderStyle.errorColor
+            if let color = style?.error?.isHidden == true ? normalColor : errorColor {
```

Comment:
Syntax its simpler now by separating the color declarations, but you still have a ternary operator used inside an if unwrapping.
`if let optional ` is one thing, but
`if let value = condition ? optional1 : optional 2` is not easy to read imo.

I would still suggest that unwrapping the `style` inside the if will provide simplest writing for this.

```suggestion
    if shouldEndEditing,
        let style = style {
                let normalColor = style.textfield.borderStyle.normalColor
                let errorColor = style.textfield.borderStyle.errorColor
                let textfieldColor = style.error?.isHidden == true ? normalColor : errorColor
                textFieldContainerBorder.updateBorderColor(to: textfieldColor)
    }
```